src/v/utils/mutex.h

```diff
@@ -48,10 +52,24 @@ class basic_mutex {
         return ss::with_semaphore(_sem, 1, std::forward<Func>(func));
     }
 
+    template<typename Func>
+    auto with(Func&& func, duration timeout) noexcept {
```

Comment:
i think the functor should come last in an interface like this

Reply:
it makes it hard to tell that a timeout is being used here, especially if the callback would be much larger.

```
        return _op_lock
          .with(
            [this, r = std::move(r)]() mutable {
                return do_vote(std::move(r));
            },
            _jit.base_duration())
```

Reply:
good idea, will change